libwacom/libwacom-database.c

```diff
@@ -279,8 +323,88 @@ parse_stylus_id(const char *str, WacomStylusId *id)
 	return ret;
 }
 
+static char *
+string_or_fallback (GKeyFile *keyfile, const char *group,
+		    const char *key, const char *fallback) {
+	g_autoptr(GError) local_error = NULL;
```

Comment:
Regarding the use of `g_autoptr` see my comment in https://github.com/linuxwacom/libwacom/issues/142#issuecomment-2290465558 - I think right now scan-build is more useful to us than g_autoptr/g_autofree so as annoying as it is we should do these manually.

I've used it in my suggestions below but only because it's easier to type out here than adding the correct free lines, so please adjust for that